tensorcircuit/results/qem/benchmark_circuits.py

```diff
@@ -0,0 +1,102 @@
+"""
+circuits for quantum chip benchmark
+"""
+
+from typing import Any, List, Dict, Tuple
+
+from mitiq.benchmarks import (
+    generate_ghz_circuit,
+    generate_mirror_circuit,
+    generate_rb_circuits,
+    generate_w_circuit,
+)
+from mitiq.interface import (
+    convert_from_mitiq,
+)
+import networkx as nx
+
+
+from ... import Circuit
+
+
+def ghz_circuit(num_qubits: int) -> Tuple[Any, Dict[str, float]]:
+    cirq = generate_ghz_circuit(num_qubits)
+    ideal = {"0" * num_qubits: 0.5, "1" * num_qubits: 0.5}
+    qisc = convert_from_mitiq(cirq, "qiskit")
+    circuit = Circuit.from_qiskit(qisc, qisc.num_qubits)
+    return circuit, ideal
+
+
+def w_circuit(num_qubits: int) -> Tuple[Any, Dict[str, float]]:
+    # Efficient quantum algorithms for GHZ and W states https://arxiv.org/abs/1807.05572
+    # Werner-state with linear complexity  {'1000': 0.25, '0100': 0.25, '0010': 0.25, '0001': 0.25}
+    cirq = generate_w_circuit(num_qubits)
+
+    ideal = {}
+    for i in range(num_qubits):
+        bitstring = "0" * i + "1" + "0" * (num_qubits - i - 1)
+        ideal[bitstring] = 1 / num_qubits
+
+    qisc = convert_from_mitiq(cirq, "qiskit")
+    circuit = Circuit.from_qiskit(qisc, qisc.num_qubits)
+    return circuit, ideal
+
+
+def rb_circuit(num_qubits: int, depth: int) -> Tuple[Any, Dict[str, float]]:
+    # num_qubits limited to 1 or 2
+    cirq = generate_rb_circuits(num_qubits, depth)[0]
+    ideal = {"0" * num_qubits: 1.0}
+    qisc = convert_from_mitiq(cirq, "qiskit")
+    circuit = Circuit.from_qiskit(qisc, qisc.num_qubits)
+    return circuit, ideal
+
+
+def mirror_circuit(
+    depth: int,
+    two_qubit_gate_prob: float,
+    connectivity_graph: nx.Graph,
+    seed: int,
+    two_qubit_gate_name: str = "CNOT",
+) -> Tuple[Any, Dict[str, float]]:
+    # Measuring the Capabilities of Quantum Computers https://arxiv.org/pdf/2008.11294.pdf
+    cirq, bitstring_list = generate_mirror_circuit(
+        nlayers=depth,
+        two_qubit_gate_prob=two_qubit_gate_prob,
+        connectivity_graph=connectivity_graph,
+        two_qubit_gate_name=two_qubit_gate_name,
+        seed=seed,
+    )
+    ideal_bitstring = "".join(map(str, bitstring_list))
+    ideal = {ideal_bitstring: 1.0}
+    qisc = convert_from_mitiq(cirq, "qiskit")
+    circuit = Circuit.from_qiskit(qisc, qisc.num_qubits)
+    return circuit, ideal
+
+
+def QAOA_circuit(
+    graph: List[Tuple[int]], weight: List[float], params: List[float]
```

Comment:
is this graph a networkx graph object or not. I suggest using network graph. But if it is the case, the annotation should be simply Any?

Reply:
This function also doesn't have the ideal results in return, will this inconsistency between other circuit generation methods incur issues?